torch/fx/passes/split_module.py

```diff
@@ -195,52 +277,68 @@ def record_cross_partition_use(
                 node.args[0], lambda n: record_cross_partition_use(n, None)
             )
             continue
-        partition_name = str(split_callback(node))
-
-        # add node to partitions
-        partition = partitions.get(partition_name)
-        if partition is None:
-            partitions[partition_name] = partition = Partition(partition_name)
 
-        partition.node_names.append(node.name)
-        node._fx_partition = partition_name
+        if assert_monotonically_increasing:
+            pid = split_callback(node)
+            assert highest_partition <= pid,\
+                ("autocast or set_grad_enabled require monotonically increasing partitions:"
+                 f"highest: {highest_partition}, this node's: {pid}")
+            highest_partition = pid
 
-        torch.fx.graph.map_arg(
-            node.args, lambda def_node: record_cross_partition_use(def_node, node)
-        )
-        torch.fx.graph.map_arg(
-            node.kwargs, lambda def_node: record_cross_partition_use(def_node, node)
-        )  # noqa: B950
+        if node.target not in SUPER_NODE_TARGETS:
```

Comment:
Add comment: do not capture cross-partition dependencies for global state nodes as their setup/unwind will be inlined into the graph.